packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -40,12 +41,18 @@ const authProvidersList: Record<string, string> = {
   'appleid.apple.com': 'appleAppId',
 };
 
+/**
+ *
+ */
 export type AmplifyAuthCognitoStackProps = {
   synthesizer: cdk.IStackSynthesizer;
 };
 
+/**
+ *
+ */
 export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCognitoStackTemplate, AmplifyStackTemplate {
-  private _scope: cdk.Construct;
+  private _scope: Construct;
```

Comment:
Few questions here while thinking about transition period and backwards compatibility.
1. Is AmplifyAuthCognitoStack a type we expose to users as public api?
2. If yes, then do we expose cdk types (like Construct here) as properties? In other words - are cdk types part of our public contract ?
3. If yes, then what happens if customer has code written against v1 and we change these props to use v2 types. Would user's project break?

The concern here is that while we could provide side by side support for internals and custom logic owned by customer we won't be able to offer "union" contract that would support both v1 and v2 types. (Unless there's some TS feature I'm not aware of that would make this happen - please correct me if I'm wrong here).

Reply:
1. Yes, Though we don't expose full class but we expose an interface ```AmplifyAuthCognitoStackTemplate```
2. We expose such interfaces with contains L1 constructs of CFN Resources.  CDK types aren't a part of public contract.
3. Since we are exposing L1 Constructs and as per migrating  docs : 
```
L1 (CfnXXXX) constructs, which represent the exact resources available in AWS CloudFormation, are always considered stable and so are included in aws-cdk-lib
```
This means what we have exposed till now is backwards compatible. It shouldn't break for categories which support override command such as auth and storage but custom category code can break for customers.

I agree we cant provide a "union contract" to support both v1 and v2 for the customers but we can find a path where the categories which are backwards compatible can use new version and the categories which aren't can have  another package to support v2 version. The switching of packages can depend on a configuration flag. We can then give some time to customers to switch to higher version before deprecating amplify packages using V1.

What do you think @sobolk ?

Reply:
Awesome, thanks for the clarification @akshbhu . That definitely reduces the scope of potential breaks.

What will happen if somebody has such `override.ts` in auth when we move `AmplifyAuthCognitoStackTemplate` members to cdk v2? I.e. will assignment `exampleObject.identityPool = resources.identityPool` compile?

```
import { AmplifyAuthCognitoStackTemplate } from '@aws-amplify/cli-extensibility-helper';
import * as cognito from '@aws-cdk/aws-cognito';

class ExampleClass {
    identityPool?: cognito.CfnIdentityPool;
}

export function override(resources: AmplifyAuthCognitoStackTemplate) {
    const exampleObject = new ExampleClass();
    exampleObject.identityPool = resources.identityPool
}

```